contracts/treasury/LivepeerGovernor.sol

```diff
@@ -0,0 +1,185 @@
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesQuorumFractionUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorSettingsUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorTimelockControlUpgradeable.sol";
+
+import "../bonding/libraries/EarningsPool.sol";
+import "../bonding/libraries/EarningsPoolLIP36.sol";
+import "../polling/PollCreator.sol";
+
+import "../ManagerProxyTarget.sol";
+import "../IController.sol";
+import "../rounds/IRoundsManager.sol";
+import "./GovernorCountingOverridable.sol";
+import "./BondingCheckpointsVotes.sol";
+
+/**
+ * @title LivepeerGovernor
+ * @notice Core contract for Livepeer governance, starting as the treasury governor.
+ * @dev If we ever add fields to this class or more extensions, make sure to add a storage gap to our custom
+ * GovernorCountingOverridable extension.
+ */
+contract LivepeerGovernor is
+    ManagerProxyTarget,
+    Initializable,
+    GovernorUpgradeable,
+    GovernorSettingsUpgradeable,
+    GovernorTimelockControlUpgradeable,
+    GovernorVotesUpgradeable,
+    GovernorVotesQuorumFractionUpgradeable,
+    GovernorCountingOverridable
+{
+    /**
+     * @notice TreasuryGovernor constructor. Only invokes constructor of base Manager contract with provided Controller.
+     * @dev This constructor will not initialize any state variables besides `controller`. The `initialize` function
+     * must be called through the proxy after construction to initialize the contract's state in the proxy contract.
+     * @param _controller Address of Controller that this contract will be registered with
+     */
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(address _controller) Manager(_controller) {
+        _disableInitializers();
+    }
+
+    /**
+     * Initializes the LivepeerGovernor instance. This requires the following contracts to have already been deployed
+     * and registered on the controller:
+     * - "Treasury" (TimelockControllerUpgradeable)
+     * - "BondingCheckpointsVotes"
+     * - "PollCreator"
+     */
+    function initialize(
+        uint256 initialVotingDelay,
+        uint256 initialVotingPeriod,
+        uint256 initialProposalThreshold
+    ) public initializer {
+        __Governor_init("LivepeerGovernor");
+        __GovernorSettings_init(initialVotingDelay, initialVotingPeriod, initialProposalThreshold);
+        __GovernorTimelockControl_init(treasury());
+
+        // The GovernorVotes module will hold a fixed reference to the votes contract. If we ever change its address we
+        // need to call the {bumpVotesAddress} function to update it in here as well.
+        __GovernorVotes_init(votes());
+
+        // Initialize with the same value from the existing polling system.
+        uint256 initialQuorum = pollCreator().QUORUM();
```

Comment:
IMO simpler to not sync with the PollCreator here and to just have independent quorum and quota state variables tracked by LivepeerGovernor giving flexibility for the parameters of the two systems to be independent one another even if they initially have the same values.

Reply:
Cool, I think it makes a lot of sense as well and I had to do additional stuff to make this work (e.g. registering the poll creator on controller) which shouldn't be necessary. 

I'll change these to be first-party parameters from the governor, with the caveat that I think the `quota()` should not be updateable for now. That because we either have to implement a full checkpointing logic [1] for the quota value or we would be subject to a similar vulnerability like this [2]. So in order to simplify this first version, I'll just make it a constant or a parameter on `initialize` for now.

[1] The simplest implementation being something [like this](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/0de6a5343637ee02455119b5c137fa33ec952705/contracts/governance/extensions/GovernorVotesQuorumFractionUpgradeable.sol) which is for the quorum. It's practically >100 additional LoC + tests and spec etc, even using the OZ checkpoints lib underneath.
[2] https://github.com/OpenZeppelin/openzeppelin-contracts/security/advisories/GHSA-xrc4-737v-9q75

Reply:
> I think the quota() should not be updateable for now

Sounds ok, but just so I understand...is LivepeerGovernor upgradeable? Meaning if we wanted to update the quota we have the ability to do so via updating the LivepeerGovernor contract itself? And if so, is the mechanism that it is owned by itself (IE upgradable via a voting action), or owned by the security committee multisig?

Reply:
Yeah, it is upgradeable! Just note that if we do upgrade it for the quota we'll need to upgrade it to have the full checkpoint history instead of just changing the current value, or we would be subject to the same vulnerability linked above for the updatable quorum.

As for how the upgrades themselves work, the governor is upgradeable through the same mechanism as our other contacts in the protocol today, through a Target contract registered in the controller which can only be changed by the controller owner. When we do make the change for the governor to own the protocol then the governor would be able to upgrade itself (as well as the rest of the protocol), but for now it's subject to our multisig.

@dob Does that make sense/sound good?

Reply:
Yes, that makes sense to me and sounds good.